doc/api/download2Go.md

```diff
@@ -0,0 +1,314 @@
+# Download2Go
+
+## Overview
+
+The Download2Go capabilities offer a full API to download a video content while online and then consuming it offline.  
+However, the different use cases are not frozen and we may use it for other cases such as low quality connection.
+
+:warning: This tool is still in an experimental phase. We are progressively releasing it in production but the
+api can still change in the future.
+
+New improvment are still pushed regularly to make it even better!  
+We are waiting your feedback to improve it!
+
+## How to use it
+
+As an experimental tool, the Download2Go won't be included in a
+default RxPlayer build.
+
+Instead, it should be imported by adding the RxPlayer through a dependency
+trough the npm registry (e.g. by doing something like `npm install rx-player`) and then specifically importing this tool from
+`"rx-player/experimental/tools"`:
+
+```js
+import { download2Go as D2G } from "rx-player/experimental/tools";
+
+this.D2G = new D2G();
+await d2g.initDB(); // is Mandatory to initialize DB
+```
+
+## Properties
+
+_nameB_ : `string` The name of the IndexDB database.
+
+## Methods
+
+#### Advanced start
+
+```js
+this.D2G = new D2G({
+  nameDB: "d2g-CompanyX"
+});
+await d2g.initDB(); // is Mandatory to initialize DB
+```
+
+You can pass 2 additionals arguments:
+
+- `nameDB`: Simply let you the possibility to name your IndexDB database :)
+  - `default`: d2g
+
+### _download()_
+
+#### Overview
+
+> The download method is the key method, she is used to retrieve and download assets while the user is online.
+
+We are be able to:
+
+- Retrieving a licence (Widevine) - Download all assets related to play a content offline (Audio/Video/Texts) - Dash (.mpd) content is currently supported (Template and SegmentSIDX are supported only) - Depending of you strategies of save, we can play a content as soon as the first segments necessary are downloaded.
+
+#### Usage
+
+`Promise: download(IAddMovie): Promise<void>`
+
+- _url_ (`string`): Mandatory property, The url of the dash manifest (must be a mpd file),
+
+- _transport_ (`smooth|dash`):` Mandatory property, tell to the loader what kind of file we will download.
+
+- _contentID_ (`string`): Mandatory property, that uniquely identify a content
+
+- _metaData_ (`any`): You can put whatever value you want here to get it in offline mode. Must be a javascript basic value type otherwise IndexDB won't insert it.
```

Comment:
`IndexedDB`

Reply:
"javascript basic value type" does not mean much. A callback is a "basic value type"